src/Model/Scope/Condition.php

```diff
@@ -195,8 +195,14 @@ public function toQueryArguments(): array
                     }
 
                     foreach (array_reverse($refModels) as $refModel) {
+                        // # is alias for count(*)
                         if ($field === '#') {
-                            $field = $value ? $refModel->action('count') : $refModel->action('exists');
+                            $field = 'count(*)';
+                        }
+
+                        // support for conditions on aggregates
+                        if (preg_match('~^(\w+)\(([\w *]*)$~', $field, $matches)) {
+                            $field = $value ? $refModel->action('fx', array_slice($matches, 1)) : $refModel->action('exists');
```

Comment:
simplified, no need to test for `(` - this will/should result in invalid column in standard addCondition

also, I do not think fallback to `exists` is correct with fx now

Reply:
Now all is adjusted

Reply:
https://github.com/atk4/data/pull/684/commits/41f3cb186fbe06e2938ed64ba6f661477bcc6818 is still overwrited - badly staged

Reply:
Your regex does not work

Reply:
probably missing `\)` before `$`

send examples of bad inputs then

Reply:
The tests are failing here because sum(price) is not detected as matching the regex

https://github.com/atk4/data/blob/5dee73a7436ebccf7ea329efd680d5b8efca7c7e/tests/ScopeTest.php#L288-L291

Reply:
yes - https://3v4l.org/4b0Yh - missing `\)`

Reply:
`is_string($field)` tested twice - remove here

Reply:
Check of `is_string($field)` is necessary as we modify `$field` within the loop

Reply:
> 
> 
> Check of `is_string($field)` is necessary as we modify `$field` within the loop

where? previously you was using str_contains so I do not think so...

Reply:
Lines 200, 208 or 216 change `$field` within the loop

https://github.com/atk4/data/blob/ec694af90dbd70fa0a977e5f8c33346f34a00bda/src/Model/Scope/Condition.php#L197-L220

Reply:
ok - 216, 200 and 208 never

Reply:
preg_match expect subject to be string. The check is necessary. Marking this as resolved